subsys/bluetooth/controller/ll_sw/ull_df.c

```diff
@@ -1189,21 +1197,73 @@ uint8_t ll_df_set_conn_cte_req_enable(uint16_t handle, uint8_t enable, uint8_t c
 		/* If controller is aware of features supported by peer device then check
 		 * whether required features are enabled.
 		 */
-		if (conn->common.fex_valid &&
-		    (!(conn->llcp_feature.features_peer & BIT64(BT_LE_FEAT_BIT_CONN_CTE_RESP)) ||
+		if (conn->llcp.fex.valid &&
+		    (!(conn->llcp.fex.features_peer & BIT64(BT_LE_FEAT_BIT_CONN_CTE_RESP)) ||
 		     ((requested_cte_type == BT_HCI_LE_AOD_CTE_1US ||
 		       requested_cte_type == BT_HCI_LE_AOD_CTE_2US) &&
-		      !(conn->llcp_feature.features_peer &
+		      !(conn->llcp.fex.features_peer &
 			BIT64(BT_LE_FEAT_BIT_ANT_SWITCH_TX_AOD))))) {
 			return BT_HCI_ERR_UNSUPP_REMOTE_FEATURE;
 		}
+
+		conn->llcp.cte_req.is_enabled = true;
+		conn->llcp.cte_req.req_interval = cte_request_interval;
+		conn->llcp.cte_req.cte_type = requested_cte_type;
+		conn->llcp.cte_req.min_cte_len = requested_cte_length;
+	}
+
+	return ull_cp_cte_req(conn, requested_cte_length, requested_cte_type);
+}
+#endif /* CONFIG_BT_CTLR_DF_CONN_CTE_REQ */
+
+#if defined(CONFIG_BT_CTLR_DF_CONN_CTE_RSP)
+/**
+ * @brief Function enables or disables CTE response control procedure for a connection.
+ *
+ * @param handle Connection handle.
+ * @param enable Enable or disable CTE response.
+ *
+ * @return HCI Status of command completion.
+ */
+uint8_t ll_df_set_conn_cte_rsp_enable(uint16_t handle, uint8_t enable)
+{
+	struct ll_conn *conn;
+
+	conn = ll_connected_get(handle);
+	if (!conn) {
+		return BT_HCI_ERR_UNKNOWN_CONN_ID;
+	}
+
+	if (!enable) {
+		/* There is no parameter validation for disable operation. */
+
+		/* TODO: Add missing implementation of disable CTE reques.
+		 * Requires refactored LLCPs.
+		 */
+	} else {
+		if (!conn->lll.df_tx_cfg.is_initialized) {
+			return BT_HCI_ERR_CMD_DISALLOWED;
+		}
+
+#if defined(CONFIG_BT_CTLR_PHY)
+		/* Phy may be changed to CODED only if PHY update procedure is supproted. In other
```

Comment:
Typo
```suggestion
		/* Phy may be changed to CODED only if PHY update procedure is supported. In other
```

Reply:
The comment was wrong actually. I've changed it.